src/app/components/Application/Application.jsx

```diff
@@ -26,16 +26,6 @@ class App extends React.Component {
     this.onChange = this.onChange.bind(this);
   }
 
-  componentWillMount() {
-    const { data } = this.state;
-    if (!data.searchResults) {
-      ajaxCall(`${appConfig.baseUrl}/api?q=${data.searchKeywords}`, (response) => {
-        Actions.updateSearchResults(response.data.searchResults);
-        Actions.updateSearchKeywords(data.searchKeywords);
-      });
-    }
-  }
-
```

Comment:
@EdwinGuzman Hi Edwin. We are trying to update code to remove lifecycle methods that React is deprecating, like `componentWillMount()`. Do you recall what the intent of this code block was?

Reply:
Refractoring code to remove lifecycle methods is nice but I wouldn't go out of our way to do so. We still do have about a year or so until those warnings become errors. In some cases, refactoring can be more involved than just removing the methods. We've been trying to refactor to more functional components and also using HOC and Discovery can benefit from that, too (but don't force it!).

In this case, I believe it deals with some bugs from the `history` package. It did not track routing like we hoped it would and clicking on the back button would not always return what we wanted. So if you searched for `Harry Potter` and got back 25 results, clicked on one, and then hit the back button in the browser _instead_ of clicking on a "return to search results" link, then the results would disappear from the store. I'm pretty sure this handles the case for the browser's back button.

There may be another case it can handle but I think this was the biggest one.

Reply:
Thanks for the explanation, Edwin! I actually still do have issues like that on production SCC. I will keep that in mind as we develop. One of the big things we want to refactor once the dependency issues are fixed is routing and history. At the very least, I am hoping to add some explanations for anywhere we decide to keep an unsafe lifecycle method or handling of refs.

Reply:
There was some work a while back to fix these routing issues so if we are still seeing problems we should definitely record that.